Add contribution guidelines for the i18n project (#353)
Joystream/joystream-org: src/locales/CONTRIBUTING.md

## Table of Contents

- [File Structure and Translation Process](#file-structure-and-translation-process)
- Contribution Process
  - [Option 1 (making the changes on your local machine and pushing the changes to GitHub)](#option-1)
  - [Option 2 (making the changes on GitHub's UI)](#option-2)
  - [Option 3 (download file locally and send changes through Discord)](#option-3)
- [Seeing the changes](#seeing-the-changes)

## File Structure and Translation Process 

The translations are written in .json files and here is an example excerpt from one of them: `(/en/sumer.json)`

```js
{
  "sumer": {
    "siteMetadata": {
      "description": "Explore the Sumer testnet"
    },
    "hero": {
      "title": "Sumer Network",
      "text": "Among other changes, the Sumer testnet release offers major improvements to content uploads through our playback app."
    },
  ...
  }
}
```
To translate this file, you would need to:
- Find the folder in `/src/locales` that corresponds to the language you want to translate to

> - Find a file in that folder (ex. `/src/locales/bs`) that has not yet been translated to that language (**check if there aren't any active PR's addressing the file as well**)

Reviewer: I think this will be quite confusing for many users. Consider adding that they can also create new folders, and how to proceed:
> - Find a file in that folder (ex. `/src/locales/bs`) that has not yet been translated to that language (**check if there aren't any active PR's addressing the file as well**)

The folders are named by the two letter [ISO language code](https://www.loc.gov/standards/iso639-2/php/code_list.php), eg. `en` means English. If you want to translate to Bosnian, `bs`, check if there is already a folder with that name. If not, copy the `en` folder, and rename it `bs`. Then create a new folder, and name it `todo`, and place all the `*.json` files in it. Once you have completed translating one file, move it back to `bs`.

Author: This is along the lines of what I initially had but to enable this and enable users to be able to see the changes they've made I need to add the language to the Gatsby config that deals with the translation. Shouldn't be too difficult but it doesn't quite allow people to arbitrarily add new translations wherever they like.

That being said I really don't have a preference for either and we should be able to add a bunch languages to the config. But even then I would have to create the associated locales folders myself and fill them with the english translations as only adding the language to the config would make Gatsby create the related pages (e.g. `${baseUrl}/ru`) but instead of words it would have the keys displayed (e.g. `foundingMembers.landing.hero.text`) as there would be no values corresponding to those keys. I'm not sure that is something we want to have on the live page (it would be hidden though and you could only access it if you were really looking for it)?

With the current approach we can have one/multiple language(s) at a time as a  "community project" and make sure that it's polished and works.

> - The translations in that file will be in english. You can now translate those values to the language corresponding to the locale (**only change the values without changing any of the keys**)

Reviewer 2: ```suggestion
- The translations in that file will be in English. You can now translate those values to the language corresponding to the locale (**only change the values without changing any of the keys**)
```

> After that is done, you will will then have something like this: `(/bs/sumer.json)`

Reviewer: Consider using full path throughout -> (`/src/locales/bs/sumer.json`)...

2. Create a branch (use descriptive names, e.g. `sumer-translation-ru`) and pull changes from the upstream `development` branch
3. Translate the file that you want to according to the previously explained rules
4. After you're done, push the changes and create a Pull Request (**make sure to write an explanatory title and add the** `internationalization` **label**)
5. In the description reference the issue pertaining to the translation and finish PR creation

### Option 2:
If you're not as familiar with git and the whole process of making contributions on GitHub, don't worry! Here is a detailed explanation to make sure that you can also contribute to our project:

**Step 1:** Fork the joystream-org repository to your own account.

![step one: fork joystream-org repo](../assets/images/i18n-readme/step-one.png)

**Step 2:** Go to your fork and switch over to the development branch.

![step two: go to your fork and switch branch to development](../assets/images/i18n-readme/step-two.png)

**Step 3:** If your branch is behind, make sure to pull the latest changes.

![step three: pull latest changes if necessary](../assets/images/i18n-readme/step-three.png)

> **Step 4:** Find the file you wish to translate (that still hasn't been translated) and create a branch that explains exactly what you wish to do. In this example we've used `sumer-translation-ru` to signify that we are translating the sumer testnet file into russian. You don't have to use this exact naming convention but make sure that it is clear what changes you've made!

Reviewer 2: ```suggestion
**Step 4:** Find the file you wish to translate (that still hasn't been translated) and create a branch that explains exactly what you wish to do. In this example we've used `sumer-translation-ru` to signify that we are translating the sumer testnet file into Russian. You don't have to use this exact naming convention but make sure that it is clear what changes you've made!
```

![step seven: write a meaningful commit message and push commit](../assets/images/i18n-readme/step-seven.png)

**Step 8:** Now you can go back to your fork and you will be greeted by a new message that will let you know that the changes you've made have indeed been "saved". You can now click *Compare & pull request*

![step eight: go back to fork and click compare and pull request](../assets/images/i18n-readme/step-eight.png)

**Step 9:** In this final step, make sure to change the branch to development and then fill out all the necessary PR metadata. Your title and description should meaningfully explain your changes (you can use the example below to get a better idea of what to do). Please don't forget to reference the issue that tracks the translations you're making and also add the `internationalization` label. And finally, after all this hard work you can press *Create pull request*. Thank you! 🎉

![step nine: change branch to dev, explain what you did, reference necessary issue, add internationalization label and create pull request](../assets/images/i18n-readme/step-nine.png)

### Option 3:

> If you're not at all familiar with git and how it works but still would love to contribute to the internationalization project, don't worry! In this last option you can download the json file and translate the necessary text as explained above in the [File Structure and Translation Process](#file-structure-and-translation-process) and send the translated text to DzideX#9084 on Discord. You will then be correctly attributed for having translated the file in the PR that will be made. Be sure to send your Testnet profile username if you wish to be awarded founding members points as well!

Reviewer 2: ```suggestion
If you're not at all familiar with git and how it works but still would love to contribute to the internationalization project, don't worry! In this last option you can download the json file and translate the necessary text as explained above in the [File Structure and Translation Process](#file-structure-and-translation-process) and send the translated text to DzhideX#9084 on Discord. You will then be correctly attributed for having translated the file in the PR that will be made. Be sure to send your testnet membership handle if you wish to be awarded founding members points as well!
```
